src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -573,11 +573,15 @@ private function getOrderReturns(Order $order): OrderReturnsForViewing
 
             if (isset($orderReturn['url'], $orderReturn['tracking_number'])) {
                 $trackingUrl = $orderReturn['url'];
-                $trackingNumber = $orderReturn['tracking_number'];
-            } elseif (isset($orderReturn['tracking_number'])) {
-                $trackingNumber = $orderReturn['tracking_number'];
             }
 
+            $conf = $this->configuration->get('PS_RETURN_PREFIX');
+            $trackingNumber = sprintf(
```

Comment:
Logic seems good in general. But if I understand correctly, actually this is not a tracking number, right ? This is a "merchandise return ID". So we should modify the variable name to make it correct, wdyt ?

Reply:
Hi @matks I agree with you but : 

After this RP I wanted to open another issue.
Which will correct the variable names and clean up this file :

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Domain/Order/QueryResult/OrderReturnForViewing.php

There are: $carrierId; $trackingUrl; $trackingNumber ;
I don't think they belong here, but it's possible that people are using them. And if I delete them, it's a break in the BC.
This fixed PR https://github.com/PrestaShop/PrestaShop/issues/19200 we need to open another one for the other problem.

Reply:
I think it should be done in the same PR. :thinking:

Reply:
I would need to know what I am deleting / rename  in[ this file](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Domain/Order/QueryResult/OrderReturnForViewing.php)? 

_There are: $carrierId; $trackingUrl; $trackingNumber ;
I don't think they belong here, but it's possible that people are using them. And if I delete them, it's a BC break._


@Progi1984 @PierreRambaud @matks ?

Reply:
For me it should be a new variable as it's not related to the tracking number

Reply:
@PierreRambaud i remove also `$carrierId; $trackingUrl; $trackingNumber ; `